zapcore/field.go

```diff
@@ -160,7 +160,7 @@ func (f Field) AddTo(enc ObjectEncoder) {
 	case NamespaceType:
 		enc.OpenNamespace(f.Key)
 	case StringerType:
-		enc.AddString(f.Key, f.Interface.(fmt.Stringer).String())
+		encodeStringer(f.Key, f.Interface, enc)
```

Comment:
rather than duplicating the logic of how errors are encoded, this should probably be `err = encodeStringer(..)` and the panic handling should return an error